File: leetcode/2582-pass-the-pillow.py
class Solution:
    def passThePillow(self, n: int, time: int) -> int:
        loop_period = (n-1)*2
        time = time % (loop_period)
        if time >= n:
            back = time - (n-1)
            return n-back 
        else:
            return 1 + time

File: src/2582-pass-the-pillow.js

/**
 * @param {number} n
 * @param {number} time
 * @return {number}
 */
var passThePillow = function (n, time) {
    let round = Math.floor(time / (n - 1))
    let movements = time % (n - 1)

    if (round % 2 === 0) { 
        return 1 + movements 
    } else {
        return n - movements
    }
};
